api/lib/domain/services/user-reconciliation-service.js

```diff
@@ -0,0 +1,67 @@
+const { normalizeAndRemoveAccents, removeSpecialCharacters } = require('../services/validation-treatments');
+const { getLevenshteinRatio, getSmallestLevenshteinRatio } = require('../services/validation-comparison');
+const _ = require('lodash');
+
+function _areTwoStringsCloseEnough(inputString, reference) {
+  return getLevenshteinRatio(inputString, reference) <= 0.25;
+}
+
+function _areTwoStringsCloseEnoughWithSeveralPossibilities(inputString, references) {
+  return getSmallestLevenshteinRatio(inputString, references) <= 0.25;
+}
+
+function _standardize(stringToStandardize) {
+  return removeSpecialCharacters(normalizeAndRemoveAccents(stringToStandardize));
+}
+
+function _findPretendersMatchingWithUser(matchingUserPretendersStandardized, standardizedUser, firstNameAlternative) {
+  return matchingUserPretendersStandardized.filter((pretender) =>
+    pretender[firstNameAlternative] && pretender.lastName
+      && _areTwoStringsCloseEnough(standardizedUser.firstName, pretender[firstNameAlternative])
+      && _areTwoStringsCloseEnoughWithSeveralPossibilities(standardizedUser.lastName, [pretender.lastName, pretender.preferredLastName])
+  );
+}
+
+function _standardizeUserAndMatchingUserPretenders(user, matchingUserPretenders) {
+  const standardizedUser = {
+    firstName: _standardize(user.firstName),
+    lastName: _standardize(user.lastName)
+  };
+
+  const matchingUserPretendersStandardized = matchingUserPretenders.map((pretender) => {
+    return {
+      id: pretender.id,
+      firstName: pretender.firstName ? _standardize(pretender.firstName) : null,
+      middleName: pretender.middleName ? _standardize(pretender.middleName) : null,
+      thirdName: pretender.thirdName ? _standardize(pretender.thirdName) : null,
+      lastName: pretender.lastName ? _standardize(pretender.lastName) : null,
+      preferredLastName: pretender.preferredLastName ? _standardize(pretender.preferredLastName) : null,
+    };
+  });
+
+  return { standardizedUser, matchingUserPretendersStandardized };
+}
+
+function findMatchingPretenderIdForGivenUser(matchingUserPretenders, user) {
+
+  const { standardizedUser, matchingUserPretendersStandardized } = _standardizeUserAndMatchingUserPretenders(user, matchingUserPretenders);
+
+  const fieldNameForMatching = ['firstName', 'middleName', 'thirdName'];
+
+  let foundPretenders = [];
+
+  _.some(fieldNameForMatching, (fieldName) => {
+    foundPretenders = _findPretendersMatchingWithUser(matchingUserPretendersStandardized, standardizedUser, fieldName);
+    return foundPretenders.length === 1;
+  });
```

Comment:
```js
['firstName', 'middleName', 'thirdName'].some((fieldName) => {
    foundPretenders = _findPretendersMatchingWithUser(matchingUserPretendersStandardized, standardizedUser, fieldName);
    return foundPretenders.length === 1;
})
```
Pas besoin d'utiliser lodash.

Reply:
Sur conseil de @jonathanperret on a remplacé ça par un `for (... of ...)`